includes/templates/class-amp-post-template.php

```diff
@@ -80,14 +80,50 @@ class AMP_Post_Template {
 	 */
 	public $post;
 
+	/**
+	 * Custom template root.
+	 *
+	 * @since 0.7
+	 * @var bool
+	 */
+	public $custom_template_dir;
+
+	/**
+	 * If template system is currently active.
+	 *
+	 * @since 0.7
+	 * @var bool
+	 */
+	public $active;
+
 	/**
 	 * AMP_Post_Template constructor.
 	 *
 	 * @param WP_Post|int $post Post.
 	 */
 	public function __construct( $post ) {
+		$this->active              = true;
+		$this->template_dir        = apply_filters( 'amp_post_template_dir', AMP__DIR__ . '/templates' );
+		$this->custom_template_dir = false;
+
+		/**
+		 * These may be configured within theme code.
+		 *
+		 * Custom template root added via theme support.
+		 * Can also further control if custom templates are applicable depending on $post.
+		 */
+		$support = get_theme_support( 'amp' );
+		if ( is_array( $support ) ) {
+			$args = array_shift( $support );
+
+			if ( ! empty( $args['template_path'] ) ) {
+				$this->custom_template_dir = $args['template_path'];
+			}
 
-		$this->template_dir = apply_filters( 'amp_post_template_dir', AMP__DIR__ . '/templates' );
+			if ( isset( $args['active_callback'] ) ) {
+				$this->active = $args['active_callback']();
```

Comment:
I think this may need to be `call_user_func( $args['active_callback'] )` for PHP 5.2.

Reply:
Nevermind! I was way wrong. This works even back in PHP 4: https://3v4l.org/TcVa0